pandas/core/generic.py

```diff
@@ -688,6 +688,92 @@ def f(x):
 
     rename.__doc__ = _shared_docs['rename']
 
+    _shared_docs['relabel'] = """
+        Assign new axes based on list-like labels
+
+        .. versionadded:: 0.20.0
+
+        Parameters
+        ----------
+        %(axes)s : list-like, optional
+            Labels to construct new axis from - number of labels
+            must match the length of the existing axis
+        copy : boolean, default True
+            Also copy underlying data
+        inplace : boolean, default False
+            Whether to return a new %(klass)s. If True then value of copy is
+            ignored.
+
+        Returns
+        -------
+        relabeled : %(klass)s (new object)
+
+        See Also
+        --------
+        pandas.NDFrame.rename
```

Comment:
prob best to add ``pandas.Series.rename, pandas.DataFrame.rename, pandas.Panel.rename`` here instead